main.cpp

```diff
@@ -12,25 +12,34 @@ int main() {
 
     // Show the network address
     const char *ip = net.get_ip_address();
-    printf("IP address is: %s\n", ip ? ip : "No IP");
+    const char *netmask = net.get_netmask();
+    const char *gateway = net.get_gateway();
+    printf("IP address: %s\n", ip ? ip : "None");
+    printf("Netmask: %s\n", netmask ? netmask : "None");
+    printf("Gateway: %s\n", gateway ? gateway : "None");
 
     // Open a socket on the network interface, and create a TCP connection to mbed.org
     TCPSocket socket;
     socket.open(&net);
-    socket.connect("developer.mbed.org", 80);
+    socket.connect("api.ipify.org", 80);
+    char *buffer = new char[256];
```

Comment:
The best place to allocate is in global memory, since the compiler can statically check for an out-of-memory (oom) situation. Both heap and stack can be overflowed at runtime, but heap allocations are easier to reason about and track. When evaluating heap size, the first step is finding where heap allocations occur. But when evaluating stack size, you usually think in terms of functions calls, not taking into account the size of the objects allocated in the functions.

For most objects allocated on the stack, the overhead is comparable to a reasonable stack frame (~64bytes). But once you introduce buffers, the stack frame can quickly grow (in this example ~256 bytes for one buffer). The user needs to consider both what functions are called while the buffer is on the stack, and where the function is called from to make sure the stack is large enough. This isn't something most users think about when writing application code.